src/material/core/common-behaviors/common-module.ts

```diff
@@ -75,7 +101,7 @@ export class MatCommonModule {
   }
 
   private _checkDoctypeIsDefined(): void {
-    if (this._document && !this._document.doctype) {
+    if (this._isCheckEnabled('doctype') && this._document && !this._document.doctype) {
```

Comment:
String access like this won't work with Google's internal use of Closure Compiler since object properties are renamed

Reply:
Does that apply to object literals as well?

Reply:
Yep

Reply:
Changed to direct property accesses.